File: couch/perl/create/couch_create.pl
#! /usr/bin/perl
#
#	couch_create.pl
#
#					Jun/23/2014
# -------------------------------------------------------
use	strict;
use	warnings;
use	utf8;
use	Encode;
use	JSON;
#
use lib '/var/www/data_base/common/perl_common';
use text_manipulate;
use uri_get;
use file_io;
#
# -------------------------------------------------------
print (encode ('utf-8',"*** 開始 ***\n"));
#
#
my $url_collection = 'http://localhost:5984/nagano';
uri_get::uri_delete_proc ($url_collection);
uri_get::uri_put_proc ($url_collection,'{}');
#
my %dict_aa = data_prepare_proc ();
#
foreach my $key (sort keys %dict_aa){
	print $key . "\t";
	my $str_json = encode_json ($dict_aa{$key});
	print $str_json . "\n";
	my $url_target = $url_collection . "/" . $key;
	uri_get::uri_put_proc ($url_target,$str_json);
	}
#
print (encode ('utf-8',"*** 終了 ***\n"));
#
# -------------------------------------------------------
sub data_prepare_proc
{
	my %dict_aa;

	%dict_aa = text_manipulate::dict_append_proc
		('t2021','長野',75842,'2005-6-22',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2022','松本',14278,'2005-9-15',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2023','上田',58641,'2005-3-17',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2024','小諸',46732,'2005-8-22',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2025','岡谷',28149,'2005-2-28',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2026','塩尻',27638,'2005-5-9',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2027','茅野',35267,'2005-4-10',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2028','飯田',81756,'2005-10-8',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2029','中野',64327,'2005-5-21',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2030','諏訪',52917,'2005-8-14',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2031','駒ヶ根',23189,'2005-7-17',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2032','佐久',72945,'2005-5-16',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2033','伊那',21957,'2005-4-12',%dict_aa);
	%dict_aa = text_manipulate::dict_append_proc
		('t2034','千曲',35871,'2005-3-21',%dict_aa);

	return	%dict_aa;
}

# -------------------------------------------------------

File: couch/ruby/create/couch_create.rb
#! /usr/bin/ruby
# -*- encoding: utf-8 -*-
#
#	couch_create.rb
#
#					Jun/20/2014
#
require 'json'
require 'restclient'
#
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
#
#
# ---------------------------------------------------------------------
def prepare_data_proc ()
dict_aa={}

dict_aa=dict_append_proc(dict_aa,'t2021',"長野",71843,"2006-4-19")
dict_aa=dict_append_proc(dict_aa,'t2022',"松本",23789,"2006-7-27")
dict_aa=dict_append_proc(dict_aa,'t2023',"上田",47251,"2006-2-8")
dict_aa=dict_append_proc(dict_aa,'t2024',"小諸",78623,"2006-1-22")
dict_aa=dict_append_proc(dict_aa,'t2025',"岡谷",91489,"2006-5-11")
dict_aa=dict_append_proc(dict_aa,'t2026',"塩尻",27915,"2006-4-17")
dict_aa=dict_append_proc(dict_aa,'t2027',"茅野",36972,"2006-9-18")
dict_aa=dict_append_proc(dict_aa,'t2028',"飯田",74921,"2006-8-15")
dict_aa=dict_append_proc(dict_aa,'t2029',"中野",81923,"2006-10-11")
dict_aa=dict_append_proc(dict_aa,'t2030',"諏訪",47598,"2006-9-12")
dict_aa=dict_append_proc(dict_aa,'t2031',"駒ヶ根",51738,"2006-7-22")
dict_aa=dict_append_proc(dict_aa,'t2032',"佐久",82514,"2006-8-17")
dict_aa=dict_append_proc(dict_aa,'t2033',"伊那",74361,"2006-12-29")
dict_aa=dict_append_proc(dict_aa,'t2034',"千曲",15672,"2006-11-21")

end
# ---------------------------------------------------------------------
#
puts "*** 開始 ***"
#
url_collection = 'http://localhost:5984/nagano'
RestClient.delete url_collection
RestClient.put url_collection, '{}'
#
dict_aa=prepare_data_proc()
#
dict_aa.each {|key, value|
	puts key
	url_target = url_collection + '/' + key
	json_out = JSON.generate (value)
	RestClient.put url_target, json_out
	}
#
puts	"*** 終了 ***"
#
# --------------------------------------------------------------------------
